### PR TITLE
Fix nested operations with related unions and interfaces

### LINKED ISSUE #3429: `nestedOperations` does not work with interfaces/unions
**Describe the bug**
Using `nestedOperations` on an interface/unions relationship does not change the schema.

**Interface**

There is not differences in the generated schema between the following type definitions:

All nested operations should be generated:
```gql
  interface Actor {
      name: String
  }

  type ActorFoo implements Actor {
      name: String
  }

  type Movie {
      id: ID
      actors: [Actor!]! @relationship(type: "HAS_ACTOR", direction: OUT)
  }
```

No nested operations should be generated:
```gql
  interface Actor {
      name: String
  }

  type ActorFoo implements Actor {
      name: String
  }

  type Movie {
      id: ID
      actors: [Actor!]! @relationship(type: "HAS_ACTOR", direction: OUT, nestedOperations: [])
  }
```

**Union**
There is not differences in the generated schema between the following type definitions:

All nested operations should be generated:
```gql
type Actor {
    name: String
}

type Director {
    name: String
}

union Person = Actor | Director

type Movie {
    id: ID
    actors: [Person!]! @relationship(type: "HAS_ACTOR", direction: OUT)
}
```

No nested operations should be generated:
```gql
type Actor {
    name: String
}

type Director {
    name: String
}

union Person = Actor | Director

type Movie {
    id: ID
    actors: [Person!]! @relationship(type: "HAS_ACTOR", direction: OUT, nestedOperations: [])
}
```

### PR DESCRIPTION
**Pull request by Liam-Doodson**
# Description

Applies the same logic used in `create-relationship-fields` for `nestedOperations` to `create-relationship-interface-fields` and `create-relationship-union-fields` so that the new argument also supports unions/interfaces

## Complexity

Complexity: Medium (sorry quite a lot of tests have been added so we don't break any edge cases when refactoring `makeAugmentedSchema` 😬)

# Issue

Closes #3429

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3465.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 48 | 53 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 85 | 84 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 55 | 74 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3300 | 3260 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2880 | 3093 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 37 | 53 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 42 | 53 | 2174 |
|  aggregations.DeeplyNestedCount | 13070331 | 13070333 | 4770 | 4891 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 151 | 200 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 63 | 151 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 147 | 170 | 3003 |
|  connections.Connection | 14082 | 14082 | 47 | 82 | 2174 |
|  connections.NestedConnection | 45466 | 45468 | 102 | 146 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 59 | 79 | 1040 |
|  connections.ConnectionWithSortAndCypher | 3284 | 3284 | 193 | 293 | 1040 |
|  create.SimpleMutation | 7 | 7 | 57 | 59 | 1 |
|  cypher-directive.TopLevelCypherDirective | 122100 | 122100 | 574 | 671 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 153619 | 153619 | 193 | 210 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 224 | 244 | 1040 |
|  delete.NestedDeleteInUpdate | 23202 | 23202 | 150 | 175 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 51 | 128 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 149 | 124 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 42 | 41 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 55 | 81 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 19 | 21 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 48 | 44 | 2174 |
|  query.QueryWhere | 9705 | 9707 | 34 | 38 | 2164 |
|  query.SimpleQueryWithNestedWhere | 9883 | 9885 | 62 | 67 | 2164 |
|  query.Nested | 10096041 | 10096041 | 10087 | 10386 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 10017 | 10031 | 2004000 |
|  query.OrFilterOnRelationships | 41423 | 41078 | 175 | 218 | 1210 |
|  query.OrFilterOnRelationshipsAndNested | 35291 | 34957 | 251 | 306 | 1210 |
|  query.QueryWithNestedIn | 13297 | 13220 | 55 | 78 | 1182 |
|  query.NestedConnectionWhere | 9834 | 9834 | 52 | 62 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10006 | 9839 | 77 | 97 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19194 | 18996 | 137 | 150 | 1618 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 169 | 201 | 1134 |
|  query.Fulltext | 80 | 80 | 35 | 37 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 47 | 54 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 63 | 104 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1491 | 1456 | 164 | 251 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 53 | 51 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 103 | 94 | 4516 |
|  sorting.SortWithTopLevelCypher | 3121 | 3121 | 119 | 194 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 79 | 65 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 57 | 61 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 413 | 452 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 294 | 280 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 109 | 110 | 2002 |

</details>
        
Old Schema Generation: 44.551s
Schema Generation: 43.624s
Old Subgraph Schema Generation: 54.320s
Subgraph Schema Generation: 54.571s

File: packages/graphql/src/schema/create-relationship-fields/create-relationship-union-fields.ts [hunk 1]
```diff
@@ -105,8 +130,30 @@ export function createRelationshipUnionFields({
...
+            };
+        }
+
+        let fieldInputFields: Record<string, string> | undefined;
+        if (
+            nestedOperations.has(RelationshipNestedOperationsOption.CONNECT) ||
+            nestedOperations.has(RelationshipNestedOperationsOption.CREATE) ||
+            connectAndCreateAndUniqueFields
+        ) {
+            fieldInputFields = {};
```
[the remark this answers is not in the document]
Comment (Liam-Doodson) [reply packages/graphql/src/schema/create-relationship-fields/create-relationship-union-fields.ts:154]: Yeah, it's intentional. This is used to store the `connect`/`create` fields for the `<source-type><field><related-type>FieldInput`. These are added later once the inputs types have been created

[the remark this answers is not in the document]
Comment (Liam-Doodson) [reply packages/graphql/src/schema/create-relationship-fields/create-relationship-union-fields.ts:154]: I've added a comment explaining this

**Review by mjfwebb — APPROVED** [review]
Great work :)
